pandas/tests/frame/test_indexing.py

```diff
@@ -2172,10 +2172,9 @@ def test_nested_exception(self):
         index[0] = ["a", "b"]
         df.index = index
 
-        try:
-            repr(df)
-        except Exception as e:
-            assert type(e) != UnboundLocalError
+        # Note: this used to be a test that any exception that _is_
```

Comment:
Should just delete altogether? Maybe missing the point but I don't see what this was testing previously

Reply:
This test was added in https://github.com/pandas-dev/pandas/pull/2201 / https://github.com/pandas-dev/pandas/issues/2200, so there was a bug in the index assignment that raised a "wrong" error when asking for the repr. But nowadays we just don't create an invalid MultiIndex. So I think we should either just remove this test, or update it to check `df.index` with the expected value.

Reply:
will remove